VideoLocker/src/main/java/org/edx/mobile/view/PresenterActivity.java

```diff
@@ -0,0 +1,58 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.edx.mobile.base.BaseFragmentActivity;
+
+public abstract class PresenterActivity<P extends Presenter<V>, V> extends BaseFragmentActivity {
+
+    protected P presenter;
+
+    @NonNull
+    abstract protected P createPresenter();
+
+    @NonNull
+    abstract protected V createView();
+
+    @Override
+    @CallSuper
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (null == presenter) {
+            presenter = getLastCustomNonConfigurationInstance();
+            if (null == presenter) {
+                presenter = createPresenter();
+            }
+        }
+    }
+
+    @Override
+    @CallSuper
+    protected void onPostCreate(Bundle savedInstanceState) {
+        super.onPostCreate(savedInstanceState);
+        presenter.attachView(createView());
+    }
+
+    @Override
+    public final P getLastCustomNonConfigurationInstance() {
+        return (P) super.getLastCustomNonConfigurationInstance();
+    }
+
+    @Override
+    public final P onRetainCustomNonConfigurationInstance() {
```

Comment:
Note that this method is deprecated, and we're supposed to use a retained Fragment for this kind of stuff instead.